lib/pipeline.ts

```diff
@@ -293,23 +294,13 @@ export class Pipeline extends Construct {
     if (props.autoBuild) {
       this.autoBuild(props.autoBuildOptions);
     }
+  }
 
-    if (props.failureNotifySlack && props.failureNotifySlack.length > 0) {
-      props.failureNotifySlack.forEach(s => {
-        new starnotifs.CfnNotificationRule(this, `FailureSlackNotification-${s.slackChannelConfigurationName}`, {
-          name: `${this.pipeline.pipelineName}-failednotifications`,
-          detailType: 'BASIC',
-          resource: this.pipeline.pipelineArn,
-          targets: [
-            {
-              targetAddress: s.slackChannelConfigurationArn,
-              targetType: 'AWSChatbotSlack',
-            },
-          ],
-          eventTypeIds: ['codepipeline-pipeline-action-execution-failed'],
-        });
-      });
-    }
+  public notifyOnFailure(notification: IPipelineNotification) {
+    notification.bind({
+      codePipeline: this.pipeline,
```

Comment:
Is `this.pipeline` not publicly accessible? If not, it probably ought to be.

After that, it sufficess to pass `this`.